src/adaptors/memory-repo/index.js

```diff
@@ -0,0 +1,74 @@
+var concat = require('concat-stream')
+var ms = require('abstract-blob-store')
+
+function BlobStore () {
+  this.store = ms()
+}
+
+BlobStore.prototype = {
+
+  /**
+   * Read a blob on a given path
+   *
+   * @param {String} key
+   * @param {Function} cb
+   * @return {ReadableStream}
+   */
+  read: function (key, cb) {
+    var store = this.store
+    var rs = store.createReadStream(key)
+
+    function onFinish (buff) {
+      cb(null, buff.toString('utf8'))
+    }
+
+    rs.on('error', cb)
+    rs.pipe(concat(onFinish))
+
+    return rs
+  },
+
+  /**
+   * Read a blob on a given path
+   *
+   * @param {String} key
+   * @param {Function} cb
+   * @return {ReadableStream}
+   */
+  readWithoutLock: function (key, cb) {
+    return this.read(key, cb)
+  },
```

Comment:
I'm not sure if we will want to use it directly, but won't hurt having them available/thought through :)